govcd/adminvdc.go

```diff
@@ -52,33 +55,273 @@ func (adminVdc *AdminVdc) Refresh() error {
 // Returns an error if the call to vCD fails.
 // API Documentation: https://vdc-repo.vmware.com/vmwb-repository/dcr-public/7a028e78-bd37-4a6a-8298-9c26c7eeb9aa/09142237-dd46-4dee-8326-e07212fb63a8/doc/doc/operations/PUT-Vdc.html
 func (adminVdc *AdminVdc) UpdateAsync() (Task, error) {
+	apiVersion, err := adminVdc.client.maxSupportedVersion()
+	if err != nil {
+		return Task{}, err
+	}
+	realFunction, ok := vdcProducerByVersion["vdc"+apiVersionToVcdVersion[apiVersion]]
```

Comment:
Rather than `realFunction`, a better name here would be `producer`

Reply:
changed